modules/axns/src/lib.rs

```diff
@@ -233,14 +231,9 @@ macro_rules! def_resource {
 
             impl $name {
                 unsafe fn deref_from_base(&self, ns_base: *mut u8) -> &$ty {
-                    unsafe extern {
-                        fn __start_axns_resource();
-                    }
-
-                    #[unsafe(link_section = "axns_resource")]
-                    static RES: $ty = $default;
+                    $crate::def_static_resource!($ty, $default);
```

Comment:
Use `def_static_resource!(RES, $ty, $default)`, and remove `res_ptr()` in the macro.